src/skins/vector/views/organisms/RoomDirectory.js

```diff
@@ -90,10 +90,20 @@ module.exports = React.createClass({
         for (var i = 0; i < rooms.length; i++) {
             var name = rooms[i].name || rooms[i].aliases[0];
             // <img src={ MatrixClientPeg.get().getAvatarUrlForRoom(rooms[i].room_id, 40, 40, "crop") } width="40" height="40" alt=""/>
+            var readLock, joinLock;
+            if (MatrixClientPeg.get().isGuest()) {
+                if (!rooms[i].world_readable) {
+                    readLock = <img src="img/warning.png" />;
+                }
+                if (!rooms[i].guest_can_join) {
+                    joinLock = <img src="img/warning.png" />;
+                }
+            }
+
```

Comment:
Not sure what you're querying on this. If you're a guest and the room is not world readable, display a warning. If the room is also not joinable, display a different kind of warning.

If the room is world readable and joinable, we don't need to display any warnings.